packages/insomnia-app/app/ui/components/settings/general.js

```diff
@@ -108,14 +130,37 @@ class General extends React.PureComponent<Props, State> {
     app.exit();
   }
 
-  async _handleFontSizeChange(el: SyntheticEvent<HTMLInputElement>) {
-    const settings = await this._handleUpdateSetting(el);
+  _getWeightsForFont(fontFamily: string): Array<number> {
+    if (fontFamily === '__NULL__') {
+      return Object.values(EDITOR_FONT_WEITGHTS);
+    } else {
+      const weightSet = new Set();
 
-    setFont(settings);
+      const fonts = fontScanner.findFontsSync({ family: fontFamily });
+      fonts.forEach(({ weight }) => {
+        weightSet.add(weight);
+      });
+
+      return [...weightSet].sort(ascendingNumberSort);
+    }
+  }
+
+  _renderFontWeightSetting(fontFamily: string) {
+    const weights = this._getWeightsForFont(fontFamily);
+
+    this.setState({
+      editorFontWeights: weights,
+    });
   }
 
   async _handleFontChange(el: SyntheticEvent<HTMLInputElement>) {
+    const currentTarget = { name: el.currentTarget.name, value: el.currentTarget.value };
```

Comment:
Prime use case for destructuring! 😄 

```suggestion
    const { currentTarget } = el;
```

Reply:
for some reason if i do it like that then actual `value` that im getting to find weights is previously selected font instead of the new one :thinking: 
do you know why it behaves like that?

Reply:
That's...strange, not sure off the top of my head, but if it doesn't work keep it like you had it. My bad! 😔